packages/nimble-components/src/select/specs/option-groups-hld.md

```diff
@@ -0,0 +1,79 @@
+# Select and Combobox option groups
+
+## Problem Statement
+
+It is not uncommon to want to categorize the options that are present in the dropdown for a `Select`/`Combobox` component. By supporting categories, users are then able to provide options that have the same display value, but under separate categories, helping avoid confusion.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Interaction Design](./IxD.md#groups)
+-   [Sectioned Combobox (issue 791)](https://github.com/ni/nimble/issues/791)
+-   [Prototype branch Storybook](https://60e89457a987cf003efc0a5b-cvmxlqwglb.chromatic.com/?path=/docs/components-select--docs) ([code](https://github.com/ni/nimble/tree/groupable-select))
+
+## Implementation / Design
+
+The HTML native `select` supports option groups with the native `optgroup` element. Configuring the `select` to use the `optgroup` takes the following form:
+
+```html
+<select name="cars">
+    <optgroup label="Swedish Cars">
+        <option value="volvo">Volvo</option>
+        <option value="saab">Saab</option>
+    </optgroup>
+    <optgroup label="German Cars">
+        <option value="mercedes">Mercedes</option>
+        <option value="audi">Audi</option>
+    </optgroup>
+</select>
+```
+
+The Nimble `Select` (and `Combobox`) will follow the same approach, and provide a new custom element, `nimble-list-option-group`, that mirrors the `optgroup` behavior/API:
+
+```html
+<nimble-select name="cars">
+    <nimble-list-option-group label="Swedish Cars">
```

Comment:
I'm curious why you opted for a `label` attribute here rather than providing a slot (either default or named) that would be for the group name? Is it because that's the way the native select is designed, or is there some other rationale?

As a note, using a slot approach might be more future proof with other features (such as adding an icon next to the text).

Reply:
Yeah, I think the slot approach is probably the right path forward. Will change.

Reply:
I think compatibility with the optgroup api surface is useful, so being able to support `label` is handy. Is there significant complexity in supporting both? i.e. us `label` if given, otherwise use a named slot?

Reply:
While I agree that compatibility with the `optgroup` API can be beneficial, I also think we should probably pick one at first, since the APIs would not conflict with one another. My current preference is the slotting approach, as that would be more consistent with other Nimble APIs, despite this suggesting that for the label we would need a named slot (which is _not_ a current paradigm I am aware of).

Ultimately I would be fine with either approach, but I'm just not sure there is a lot of value in supporting both initially. I can be convinced otherwise and can likely be swayed with a gentle breeze.

Reply:
I'd prefer to not wait for the "use case of compatibility" so if we just are supporting one initially I'd go for label. And I'll advocate that supporting both label and slot is probably not expensive (at least slot is probably the more expensive of the two).

And agreed that the default slot would be more inline with our other apis 👍

Reply:
So, to be clear, I think if we were to take the dual-approach (or just the slot approach), the slot approach would be to use a named slot. I'm not sure how the default slot approach would work:

Default slot approach:
```html
<nimble-list-option-group>
    Group Name
    <nimble-list-option>Option 1</nimble-list-option>
    <nimble-list-option>Option 2</nimble-list-option>
<nimble-list-option-group>
```
I suppose the above would be feasible with some slot forwarding of the `nimble-list-option` elements (into a named slot in the `Select` template?), but is this a reasonable API for a client, versus:

Named slot approach:
```html
<nimble-list-option-group>
    <span slot="label">Group Name</span>
    <nimble-list-option>Option 1</nimble-list-option>
    <nimble-list-option>Option 2</nimble-list-option>
<nimble-list-option-group>
```
I suppose it won't take much effort to vet either approach out, so I guess my question is, which do we prefer?